system-parachains/encointer/src/lib.rs

```diff
@@ -0,0 +1,1005 @@
+		EncointerCommunities: pallet_encointer_communities::{Pallet, Call, Storage, Config<T>, Event<T>} = 62,
+		EncointerBalances: pallet_encointer_balances::{Pallet, Call, Storage, Config<T>, Event<T>} = 63,
+		EncointerBazaar: pallet_encointer_bazaar::{Pallet, Call, Storage, Event<T>} = 64,
+		EncointerReputationCommitments: pallet_encointer_reputation_commitments::{Pallet, Call, Storage, Event<T>} = 65,
+		EncointerFaucet: pallet_encointer_faucet::{Pallet, Call, Storage, Config<T>, Event<T>} = 66,
+	}
+}
+
+/// `parachains_common` is an upstream crate, where they are started to define common types.
+///
+/// The re-export is added by encointer.
+pub use parachains_common::{AccountId, Balance, BlockNumber, Hash, Header, Nonce, Signature};
+
+/// The address format for describing accounts.
+pub type Address = sp_runtime::MultiAddress<AccountId, ()>;
+/// Block type as expected by this runtime.
+pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+/// A Block signed with a Justification
+pub type SignedBlock = generic::SignedBlock<Block>;
+/// BlockId type as expected by this runtime.
+pub type BlockId = generic::BlockId<Block>;
+/// The SignedExtension to the basic transaction logic.
+pub type SignedExtra = (
+	frame_system::CheckNonZeroSender<Runtime>,
+	frame_system::CheckSpecVersion<Runtime>,
+	frame_system::CheckTxVersion<Runtime>,
+	frame_system::CheckGenesis<Runtime>,
+	frame_system::CheckEra<Runtime>,
+	frame_system::CheckNonce<Runtime>,
+	frame_system::CheckWeight<Runtime>,
+	pallet_asset_tx_payment::ChargeAssetTxPayment<Runtime>,
+);
+/// Unchecked extrinsic type as expected by this runtime.
+pub type UncheckedExtrinsic =
+	generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
+/// Extrinsic type that has already been checked.
+pub type CheckedExtrinsic = generic::CheckedExtrinsic<AccountId, RuntimeCall, SignedExtra>;
+
+/// Migrations to apply on runtime upgrade.
+pub type Migrations = (
+	// fixing the scheduler with a local migration is necessary because we have missed intermediate
+	// migrations. the safest migration is, therefore, to clear all storage and bump StorageVersion
+	migrations_fix::scheduler::v4::MigrateToV4<Runtime>,
+	// also here we're actually too late with applying the migration. however, the migration does
+	// work as-is.
+	pallet_xcm::migration::v1::VersionUncheckedMigrateToV1<Runtime>,
+	// balances are more tricky. We missed to do the migration to V1 and now we have inconsistent
+	// state which can't be decoded to V0, yet the StorageVersion is V0.
+	// the strategy is to: just pretend we're on V1
+	migrations_fix::balances::v1::BruteForceToV1<Runtime>,
+	// then reset to V0
+	pallet_balances::migration::ResetInactive<Runtime>,
+	//then apply the proper migration as we should have done earlier
+	pallet_balances::migration::MigrateToTrackInactive<Runtime, xcm_config::CheckingAccount>,
+);
+
```

Comment:
Okay thanks. It is rather simple to run the try-runtime check locally, in case you need to debug it (github is a bit slow currently).  

https://github.com/paritytech/try-runtime-cli example here https://github.com/polkadot-fellows/runtimes/blob/98872cb5cdcc65369a0d48c6b2bab95d1be5d286/.github/workflows/check-migrations.yml#L79

Reply:
yes and no. it runs through and succeeds, but it complains about a corrupt state for one key:
```
error: [2024-01-09T14:08:47Z ERROR frame_support::storage] old key failed to decode at [227, 143, 24, 82, 7, 73, 138, 187, 92, 33, 61, 15, 176, 89, 179, 216, 216, 55, 28, 116, 117, 23, 21, 29, 134, 63, 146, 111, 129, 151, 24, 223, 191, 178, 127, 30, 174, 240, 107, 185, 3, 0, 0, 0, 96, 15, 180, 143, 173, 7, 149, 151, 21, 112, 71, 148, 112, 212, 184, 115, 3, 1, 0]

converting key to hex: 0xE38F185207498ABB5C213D0FB059B3D8D8371C747517151D863F926F819718DFBFB27F1EAEF06BB903000000600FB48FAD0795971570479470D4B873030100

looking up onchain raw storage
value hex: 0x600a000000000000000003000000
```
hard to guess what key and what value that is but I'll investigate

Reply:
Looks like something in `PolkadotXcm::VersionNotifyTargets`.

Reply:
good spot. I can confirm that!

![image](https://github.com/polkadot-fellows/runtimes/assets/10366175/12ca6650-64ca-4d34-8d85-4bd82756f137)

If 2656 is a paraid, then there is no such parachain and we need not worry I guess

Reply:
I will ask in the XCM chat what this means since i have on clue.

Reply:
thank you. if we need to take action and modify the migrations, I suggest to let that be a separate PR.

Reply:
Cleanest way is to remove this key altogether, however this would mean that the relay chain would not be notified of any XCM version changes on Encointer. To remedy this, the relay chain would have to resend a `SubscribeVersion` instruction to Encointer.